src/components/panels/hours-panel-date-viewer.js

```diff
@@ -0,0 +1,586 @@
+                          {day.getDate()}
+                        </div>
+                      );
+                    })}
+                  </a>
+                );
+              })}
+            </div>
+            <p css={{ color: `var(--color-neutral-300);` }}>
+              <span css={{ background: 'white',
+                border: `solid 1px var(--color-neutral-200)`,
+                borderRadius: '4px',
+                boxShadow: `0 1px 1px rgba(0, 0, 0, .2), 0 2px 0 0 rgba(255, 255, 255, .7) inset;`,
+                display: 'inline-block',
+                fontFamily: 'monospace',
+                fontSize: '0.85rem',
+                marginBottom: SPACING.S,
+                padding: `0 ${SPACING['2XS']}` }}
+              >
+                esc
+              </span>
+              {' '}
+              to
+              close
+            </p>
+          </>
+        </div>
+      )
+    : null;
+};
+
+CalendarView.propTypes = {
+  isVisible: PropTypes.bool,
+  weekOffset: PropTypes.number
+};
+
+const HoursPanelNextPrev = ({ toggleCalendarVisibility, isCalendarVisible }) => {
+  const [{ weekOffset }, dispatch] = useStateValue();
+  const date = new Date(new Date().toLocaleString('en-US', { timeZone: 'America/New_York' }));
+
+  const fromDate = new Date(date);
+  fromDate.setDate(date.getDate() + weekOffset * 7 - date.getDay());
+
+  const toDate = new Date(date);
+  toDate.setDate(date.getDate() + weekOffset * 7 + (6 - date.getDay()));
+
+  const hoursRange = {
+    label: `Showing hours from ${dateFormat(fromDate)} to ${dateFormat(toDate)}`,
+    text: `${dateFormat(fromDate, true)} - ${dateFormat(toDate, true)}`
+  };
+
+  const handleInteraction = (event, action) => {
+    if (event.type === 'click' || (event.type === 'keydown' && (event.key === 'Enter' || event.key === ' '))) {
+      event.preventDefault();
+      action();
+    }
+  };
+
+  return (
+    <Margins
+      data-hours-panel-next-previous
+      css={{
+        position: 'relative'
+      }}
+    >
+      <div
+        css={{
+          alignItems: 'baseline',
+          display: 'flex',
+          justifyContent: 'space-between'
+        }}
+      >
+        <PreviousNextWeekButton
+          onClick={(event) => {
+            return updateWeekOffset({ dispatch, event, relativeOffset: -1, weekOffset });
+          }}
+          onKeyDown={(event) => {
+            return updateWeekOffset({ dispatch, event, relativeOffset: -1, weekOffset });
+          }}
+          type='previous'
+
+        >
+          Previous week
+        </PreviousNextWeekButton>
+        <button
+          css={isCalendarVisible
+            ? {
+                boxShadow: 'inset 0 -2px var(--color-teal-400);',
+                color: 'inherit'
+              }
+            : {
+                '&:hover': {
+                  boxShadow: 'inset 0 -2px var(--color-teal-400);'
+                },
+                boxShadow: 'none',
+                color: 'inherit'
+              }}
```

Comment:
Another opportunity to reduce code:
```suggestion
          css={{
  boxShadow: 'none',
  color: 'inherit',
  ...(isCalendarVisible
    ? {
        boxShadow: 'inset 0 -2px var(--color-teal-400)'
      }
    : {
        '&:hover': {
          boxShadow: 'inset 0 -2px var(--color-teal-400)'
        }
      })
}}
```